beacon-chain/blockchain/block_processing.go

```diff
@@ -33,6 +33,15 @@ type BlockProcessor interface {
 	CleanupBlockOperations(ctx context.Context, block *pb.BeaconBlock) error
 }
 
+// BlockFailedProcessingErr represents a block failing a state transition function.
+type BlockFailedProcessingErr struct {
```

Comment:
why does this need to be a struct ? we can use a function for this, where you input the error, and it adds the block error message to the inputted string

Reply:
This is so it implements the error type - see https://programming.guide/go/create-error.html

Reply:
Why isn’t this errors.New?